.github/workflows/main.yml

```diff
@@ -0,0 +1,272 @@
+          key: lnd-${{ runner.os }}-go-${{ env.GO_VERSION }}-${{ github.job }}-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            lnd-${{ runner.os }}-go-${{ env.GO_VERSION }}-${{ github.job }}-${{ hashFiles('**/go.sum') }}
+            lnd-${{ runner.os }}-go-${{ env.GO_VERSION }}-${{ github.job }}-
+            lnd-${{ runner.os }}-go-${{ env.GO_VERSION }}-
+            lnd-${{ runner.os }}-go-
+
+      - name: setup go ${{ env.GO_VERSION }}
+        uses: actions/setup-go@v2
+        with:
+          go-version: '~${{ env.GO_VERSION }}'
+
+      - name: bitcoin cache
+        uses: actions/cache@v1
+        with:
+          path: ~/bitcoin/bitcoin-${{ env.BITCOIN_VERSION }}/bin
+          key: lnd-${{ runner.os }}-bitcoin-${{ env.BITCOIN_VERSION }}
+          restore-keys: |
+            lnd-${{ runner.os }}-bitcoin-${{ env.BITCOIN_VERSION }}
+
+      - name: install bitcoind
+        run: ./scripts/install_bitcoind.sh
+
+      - name: run ${{ matrix.unit_type }}
+        run: make ${{ matrix.unit_type }}
+
+      - name: Send coverage
+        uses: shogo82148/actions-goveralls@v1
+        if: matrix.unit_type == 'btcd unit-cover'
+        with:
+          path-to-profile: coverage.txt
+          parallel: true
+
+  ########################
+  # run integration tests
+  ########################
+  integration-test:
+    name: run integration tests
+    runs-on: ubuntu-latest
+    strategy:
+      # Allow other tests in the matrix to continue if one fails.
+      fail-fast: false
+      matrix:
+        backend:
+          - btcd
+          - bitcoind
+          - neutrino
+    steps:
+      - name: git checkout
+        uses: actions/checkout@v1
+
+      - name: go cache
+        uses: actions/cache@v1
+        with:
+          path: /home/runner/work/go
+          key: lnd-${{ runner.os }}-go-${{ env.GO_VERSION }}-${{ github.job }}-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            lnd-${{ runner.os }}-go-${{ env.GO_VERSION }}-${{ github.job }}-${{ hashFiles('**/go.sum') }}
+            lnd-${{ runner.os }}-go-${{ env.GO_VERSION }}-${{ github.job }}-
+            lnd-${{ runner.os }}-go-${{ env.GO_VERSION }}-
+            lnd-${{ runner.os }}-go-
+
+      - name: setup go ${{ env.GO_VERSION }}
+        uses: actions/setup-go@v2
+        with:
+          go-version: '~${{ env.GO_VERSION }}'
+
+      - name: bitcoin cache
+        uses: actions/cache@v1
+        with:
+          path: ~/bitcoin/bitcoin-${{ env.BITCOIN_VERSION }}/bin
+          key: lnd-${{ runner.os }}-bitcoin-${{ env.BITCOIN_VERSION }}
+          restore-keys: |
+            lnd-${{ runner.os }}-bitcoin-${{ env.BITCOIN_VERSION }}
+
+      - name: install bitcoind
+        run: bash ./scripts/install_bitcoind.sh
+        if: matrix.backend == 'bitcoind'
+
+      - name: ${{ matrix.backend }} integration
+        run: make itest backend=${{ matrix.backend }}
+        continue-on-error: true
+
+      - name: package logfiles
```

Comment:
From inspecting the existing runs you linked it seems that the `package logfiles` and `upload logfiles` part didn't run?

Reply:
It takes almost half a minute to upload the log files. That's why I added the `if: failure()` condition to these two steps. I think we only really need the log files in case the tests fail.